### PR TITLE
Add Document about the Graph Optimization in TensorFlow

### PR DESCRIPTION
**Pull request by nuka137**
This pull request adds the document about the graph optimization performing internally in TensorFlow.
This document shows what optimizations are performed in TensorFlow, and how to enable/disable these optimizations from Python API.
I think most of TensorFlow beginners does not interests, but this document helps the advanced user who wants to control the graph optimizations manually.

This pull request is related to #511 .
#511 targets to TensorFlow 1.x, so I rewrote whole documents for TensorFlow 2.0.

### CODE REVIEW

**Comment by nuka137** [comment]
@lamberta 

I rewrote the document in #511 which is rejected due to the transition phase of the TensorFlow 1.x -> 2.x.
Is this document still acceptable in TensorFlow 2.0?

This pull request adds the new `.ipynb` file because there is no related documents about the graph optimization in the current documents.
Should I merge this document to another file?

**Comment by lamberta** [comment]
Thanks, @nuka137 !

Colab preview: https://colab.research.google.com/github/nuka137/docs/blob/graph_optimization/site/en/guide/graph_optimization.ipynb

The challenge will be finding reviewers (and it's a holiday week in the US). Looping in @mdanatg and @alextp to take a quick look. Either as a standalone user guide on tensorflow.org, or in an API reference section.

Also, for notebooks, make sure the format matches the [TF notebook template](https://github.com/tensorflow/docs/blob/master/tools/templates/notebook.ipynb).

**Comment by mdanatg** [comment]
Nice! I gave it a quick look and I think it would make a great guide,
heping to demystify what happens under the hood. Happy to review the PR
when ready.

For location, I think Tutorials/Advanced/Customization is a good place - it
seems closely related to "Performance with tf.function" that also sits in
there.

Cheers,
Dan

**Comment by nuka137** [comment]
@lamberta @mdanatg @alextp 

Thanks for your instruction.
I fixed the `graph_optimization.ipynb` to match the TF notebook template, and move to `site/en/tutorials/customization/graph_optimization.ipynb` from `site/en/guide/graph_optimization.ipynb`.
So, new link for Colab is now located at [site/en/tutorials/customization/graph_optimization](https://colab.research.google.com/github/nuka137/docs/blob/graph_optimization/site/en/tutorials/customization/graph_optimization.ipynb).

Could you review it again?

File: site/en/tutorials/customization/graph_optimization.ipynb [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+    "    return a\n",
+    "\n",
+    "print(assert_func())"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "* Disable All Graph Optimizations\n",
```
Comment (alextp) [inline site/en/tutorials/customization/graph_optimization.ipynb]: This is not getting rendered right, I think

Comment (nuka137) [reply site/en/tutorials/customization/graph_optimization.ipynb]: Thanks. I fixed to use h3 for emphasizing.

**Comment by lamberta** [comment]
Thanks @nuka137 for the updates, and @alextp for the review.

@anirudh161 if you have some time, can you take an editorial pass through this community guide and get it ready to publish on tensorflow.org?

**Comment by lamberta** [comment]
Staging link: https://colab.research.google.com/github/nuka137/docs/blob/graph_optimization/site/en/guide/_graph_optimization.ipynb

Thanks, all. Bringing this PR in so @anirudh161 can work on it before publishing

**Pull request merged**